libethereum/EthereumHost.cpp

```diff
@@ -576,8 +576,20 @@ void EthereumHost::onPeerNewBlock(EthereumPeer* _peer, RLP const& _r)
 				u256 difficulty = _r[1].toInt<u256>();
 				if (m_syncingTotalDifficulty < difficulty)
 				{
+					BlockInfo bi;
+					RLP root(_r[0].data());
+					try
+					{
+						bi.populateFromHeader(root[0], CheckEverything, h);
+					}
+					catch (Exception const& _e)
+					{
+						_peer->disable("Malformed block received.");
+						return;
+					}
+					
 					_peer->m_latestHash = h;
-					_peer->m_totalDifficulty = difficulty;
+					_peer->m_totalDifficulty = bi.difficulty;
```

Comment:
_peer->m_totalDifficulty is supposed to be a sum of difficulties for all blocks in the peer's blockchain.